storage/oasis/nodeapi/history/history.go

```diff
@@ -20,21 +20,31 @@ var _ nodeapi.ConsensusApiLite = (*HistoryConsensusApiLite)(nil)
 
 type APIConstructor func(ctx context.Context, chainContext string, archiveConfig *config.ArchiveConfig, fastStartup bool) (nodeapi.ConsensusApiLite, error)
 
+var DamaskAPIConstructor = func(ctx context.Context, chainContext string, archiveConfig *config.ArchiveConfig, fastStartup bool) (nodeapi.ConsensusApiLite, error) {
+	sdkConn, err := connections.SDKConnect(ctx, chainContext, archiveConfig.ResolvedConsensusNode(), fastStartup)
+	if err != nil {
+		return nil, err
+	}
+	return damask.NewDamaskConsensusApiLite(sdkConn.Consensus()), nil
+}
+
+var CobaltAPIConstructor = func(ctx context.Context, chainContext string, archiveConfig *config.ArchiveConfig, fastStartup bool) (nodeapi.ConsensusApiLite, error) {
+	rawConn, err := connections.RawConnect(archiveConfig.ResolvedConsensusNode())
+	if err != nil {
+		return nil, fmt.Errorf("indexer RawConnect: %w", err)
+	}
+	return cobalt.NewCobaltConsensusApiLite(rawConn), nil
+}
+
+// Map each hardcoded, indexer-specific archive name to the API constructors
+// that can talk to that archive. The namespace of archive names is shared
+// between mainnet and testnet for simplicity.
 var APIConstructors = map[string]APIConstructor{
-	"damask": func(ctx context.Context, chainContext string, archiveConfig *config.ArchiveConfig, fastStartup bool) (nodeapi.ConsensusApiLite, error) {
-		sdkConn, err := connections.SDKConnect(ctx, chainContext, archiveConfig.ResolvedConsensusNode(), fastStartup)
-		if err != nil {
-			return nil, err
-		}
-		return damask.NewDamaskConsensusApiLite(sdkConn.Consensus()), nil
-	},
-	"cobalt": func(ctx context.Context, chainContext string, archiveConfig *config.ArchiveConfig, fastStartup bool) (nodeapi.ConsensusApiLite, error) {
-		rawConn, err := connections.RawConnect(archiveConfig.ResolvedConsensusNode())
-		if err != nil {
-			return nil, fmt.Errorf("indexer RawConnect: %w", err)
-		}
-		return cobalt.NewCobaltConsensusApiLite(rawConn), nil
-	},
+	// mainnet
+	"damask": DamaskAPIConstructor,
+	"cobalt": CobaltAPIConstructor,
+	// testnet
+	"2022-03-03": DamaskAPIConstructor,
```

Comment:
huh i never noticed we named testnets just by date

Reply:
mainnet releases are by date too, at least on the mainnet-artifacts repo

Reply:
yeah oasis-core does that already. they don't deserve zingy codenames :)